src/slack.ts

```diff
@@ -1059,24 +1251,133 @@ ${usernames.join(', ')}`
+
+                    for (const pattern of listener.patterns) {
+                        const matches = text.match(pattern)
+                        if (!matches) continue
+
+                        this.log.debug(
+                            `Found match for pattern: ${pattern.source}`
+                        )
                         this.handleMatch(listener, {
                             ...chessterMessage,
-                            text: text.trim(),
                             matches,
                         })
-                        return true
-                    })
+                        break
+                    }
                 })
             } catch (error) {
-                this.log.error(
-                    `Uncaught error in handling an rtm message: ${JSON.stringify(
-                        error
-                    )}`
+                this.log.error(`Error handling message: ${error}`)
+                await say(
+                    'Error processing your message. Please try again later.'
+                )
+            }
+        })
+
+        // Set up event handler for @mentions
+        // Messages that don't ping chesster directly are handled in `this.app.message...`
+        // TODO this.app.event and this.app.message have very similar logic; combine them into a util of some sort
+        this.app.event('app_mention', async ({ event, say }) => {
+            try {
+                const channel = await this.getChannel(event.channel)
+                if (!channel) {
+                    this.log.warn(
+                        `Unable to get details for channel: ${event.channel}`
+                    )
+                    return
+                }
+
+                let text = event.text || ''
+
+                this.log.info(
+                    `Original text: "${text}", controller id: ${
+                        this.controller?.id || 'undefined'
+                    }`
                 )
-                this.log.error(`Stack: ${new Error().stack}`)
+
+                if (this.controller?.id) {
+                    text = text.replace(
+                        new RegExp(`<@${this.controller.id}>\\s*`, 'g'),
+                        ''
+                    )
+                    this.log.info(`Text after removing mention: "${text}"`)
+                }
+
+                const chessterMessage: ChessterMessage = {
+                    type: 'message',
+                    user: event.user || '',
+                    channel,
+                    text,
+                    ts: event.ts,
+                    attachments: [],
+                    isPingModerator: false,
+                }
+
+                // Loop through ALL command listeners to find matches
+                let matched = false
+
+                // Find command listeners first
+                for (const listener of this.listeners) {
+                    if (
+                        listener.type === 'command' &&
```

Comment:
Minor nit: if you invert the condition in this if, and you use `continue` instead, you can un-indent the following code and it makes things easier to understand